test/integration/testPutGet.js

```diff
@@ -857,3 +861,286 @@ describe('PUT GET test multiple files', function () {
+
+  const tmpdirPath = getPlatformTmpPath(tmpDir);
+  const getQuery = `GET ${stage} ${tmpdirPath}`;
+
+  it('testNoCompress', function (done) {
+    async.series(
+      [
+        function (callback) {
+          connection.execute({
+            sqlText: putQuery,
+            complete: function (err, stmt) {
+              if (err) {
+                callback(err);
+              } else {
+                const stream = stmt.streamRows();
+                stream.on('error', function (err) {
+                  callback(err);
+                });
+                stream.on('data', function (row) {
+                  // Check the file is correctly uploaded
+                  assert.strictEqual(row['status'], UPLOADED);
+                  assert.strictEqual(row.targetSize, ROW_DATA_SIZE);
+                });
+                stream.on('end', function () {
+                  callback();
+                });
+              }
+            }
+          });
+        },
+        function (callback) {
+          connection.execute({
+            sqlText: getQuery,
+            complete: function (err, stmt) {
+              if (err) {
+                callback(err);
+              } else {
+                const stream = stmt.streamRows();
+                stream.on('error', function (err) {
+                  callback(err);
+                });
+                stream.on('data', function (row) {
+                  assert.strictEqual(row['status'], DOWNLOADED);
+                  // Verify if the file is not compressed.
+                  const file = path.join(tmpDir, row.file);
+                  const data = fs.readFileSync(file).toString();
+                  assert.strictEqual(data, ROW_DATA);
+                });
+                stream.on('end', function () {
+                  callback();
+                });
+              }
+            }
+          });
+        }
+      ],
+      done
+    );
+  });
+});
+
+describe('PUT GET test with different size', function () {
+  let connection;
+  const stage = `@${DATABASE_NAME}.${SCHEMA_NAME}.%${TEMP_TABLE_NAME}`;
+  const createTable = `create or replace table ${TEMP_TABLE_NAME} (${COL1} STRING, ${COL2} STRING, ${COL3} STRING)`;
+  const removeFile = `REMOVE ${stage}`;
+  const dropTable = `DROP TABLE IF EXISTS ${TEMP_TABLE_NAME}`;
+
+  // Create a temp file without specified file extension
+  const zeroByteFile = tmp.fileSync();
+  const largeFile = tmp.fileSync(); 
+  // Create a tmp folder for downloaded files
+  const tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'get'));
+
+  const createEmptyFileOfSize = (fileName, size) => {
+    return new Promise((resolve) => {
+      const fh = fs.openSync(fileName, 'w');
+      fs.writeSync(fh, 'ok', Math.max(0, size - 2));
+      fs.closeSync(fh);
+      resolve(true);
+    });
+  };
+
+  before(async () => {
+    // Create a file of 100 MB
+    createEmptyFileOfSize(largeFile.name, 100*1024*1024);
```

Comment:
this function returns Promise and we are not waring for it to be finished

Reply:
updated.